### PR TITLE
Story TG-244

### PR DESCRIPTION
**Pull request by sergii-trotsiuk**
Should be discussed, because of moving shared deps to core.

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/garage-it/SmartHouse-frontend/pull/133" target="_blank">ReviewNinja

File: src/app/core/auth/auth.interfaces.ts [hunk 1]
```diff
@@ -0,0 +1,4 @@
+export interface IUserCredentials {
+    email:string,
```
Comment (yakimko) [inline src/app/core/auth/auth.interfaces.ts]: please add a space after properties

File: src/app/core/header/header.component.test.ts [hunk 2]
```diff
@@ -1,16 +1,18 @@
...
     headerName: 'Dashboard',
     iconImage: './assets/dashboard.svg',
     main: true
 };
 const NAVIGATION_ROUTES = [
     MAIN_ROUTE,
-    { path: 'devices', headerName: 'Devices', iconImage: './assets/deviceList.svg' },
-    { path: 'scenarios', headerName: 'Scenarios', iconImage: './assets/scenarioList.png' },
-    { path: 'help', headerName: 'Help', iconImage: './assets/help.svg' }
+    { path: '/devices', headerName: 'Devices', iconImage: './assets/deviceList.svg' },
```
Comment (yakimko) [inline src/app/core/header/header.component.test.ts]: maybe, we should use conts for routes from  './config.routing' ?

Comment (sergii-trotsiuk) [reply src/app/core/header/header.component.test.ts]: Agree

File: src/app/core/profile/profile.service.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+            return Promise.resolve();
+        }
+
+        this.http.setAuthHeader(token);
+
+        return this.http.get('/user/current-user').toPromise()
+            .then(user => {
+                this.setUserData(user);
+
+                return undefined;
```
Comment (yakimko) [inline src/app/core/profile/profile.service.ts]: why do you need this?

File: src/app/core/profile/profile.service.ts [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+            .catch((err) => {
+                if (err.status === 401) {
+                    this.removeUserData();
+                }
+            });
+    }
+
+    setUserData(user: User, token?: string): void {
+        this.user = user;
+        if (token) {
```
Comment (yakimko) [inline src/app/core/profile/profile.service.ts:39]: please add a blank line before

File: src/app/core/header/header.component.test.ts (hunk 2 above)
Comment (yakimko) [inline src/app/core/header/header.component.test.ts]: we need to use them through the application

File: src/app/core/profile/profile.interfaces.ts [hunk 5]
```diff
@@ -0,0 +1,7 @@
+export interface User {
+    _id: string;
+    email?: string;
+    login: string;
```
Comment (AstakhovArtem) [inline src/app/core/profile/profile.interfaces.ts:3]: login field was removed from mongo schema

File: src/app/login/login.component.ts [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+        email: null,
+        password: null
+    };
+    private loggingError: boolean = false;
+
+    constructor(private auth: AuthService, private router: Router) {}
+
+    login() {
+        this.auth.login(this.credentials)
+            .subscribe(() => {
```
Comment (AstakhovArtem) [inline src/app/login/login.component.ts:24]: If you subscribe to event stream, you should unsubscribe when component destroyed

File: src/app/core/profile/profile.service.ts (hunk 3 above)
Comment (sergii-trotsiuk) [reply src/app/core/profile/profile.service.ts]: Removed

**Pull request merged**